src/Spice86.Core/Emulator/VM/Clock/CyclesClock.cs

```diff
@@ -3,7 +3,42 @@
 using Spice86.Core.Emulator.CPU;
 
 public class CyclesClock(State cpuState, long cyclesPerSecond) : IEmulatedClock {
+    private DateTime _pauseStartTime;
+    private TimeSpan _totalPausedTime = TimeSpan.Zero;
+    private bool _isPaused;
+
     public long CyclesPerSecond { get; set; } = cyclesPerSecond;
 
-    public double CurrentTimeMs => (double)cpuState.Cycles * 1000 / CyclesPerSecond;
+    public double CurrentTimeMs {
```

Comment:
This implementation does not need to take care of the pause because when CPU is paused, cycles are stopped and time does not pass, which is the expected behaviour.